src/app/SafeAttributePersistenceProvider.h

```diff
@@ -54,7 +54,14 @@ class SafeAttributePersistenceProvider
     {
         uint8_t value[sizeof(T)];
         auto w = Encoding::LittleEndian::BufferWriter(value, sizeof(T));
-        w.EndianPut(uint64_t(aValue), sizeof(T));
+        if (std::is_signed_v<T>)
```

Comment:
This will need to be `if constexpr` to compile on Android, which means this is blocked on Ameba updating to c++17.

Reply:
What do we do here? Should I split it into two templated functions, one for signed and one for unsigned?

Reply:
We have a direct line to Ameba if you think we should ask them to support c++17.

Reply:
@andy31415 already asked @pankore a few days ago....  So now we are just waiting on them to do it.

For now, I think we can wait on landing this PR until they do that.